### DISCUSSION TITLE
Generate vercel.json build time

### DISCUSSION THREAD

**Discussion by seloner** [vercel/vercel#8137] [Help]
I want to generate to vercel.json file during the build time . I am trying with the custom build command
`require('dotenv').config();
const fs = require('fs');

const vercelJSON = {
  rewrites: [
    {
      source: '/api',
      destination: process.env.API,
    },
    {
      source: '/customapi',
      destination: process.env.CUSTOM_API,
    },
  ],
};
const JSONstring = JSON.stringify(vercelJSON, null, 2);
fs.writeFile('./vercel.json', JSONstring, { flag: 'w+' }, (err) => {
  console.log(JSONstring);
  if (err) process.exit(1);
  process.exit(0);
});
`
Should I save the file in root of the project ?

**Comment by amyegan** [comment]
Hi @seloner.  Yes, the `vercel.json` config should be in the root of the project.

**Reply by seloner** [reply]
Should be possible to do that during the build step ? I tried in the build command or in the ignore command(git settings) but it is not working .
I placed the vercel.json in the root direcotroy

**Reply by amyegan** [reply]
If you're using Next.js, [using environment variables it in your next.config.js](https://github.com/vercel/vercel/discussions/4351#discussioncomment-13370) might be an easier option. For other frameworks and libraries, I've seen people use [pre & post scripts](https://docs.npmjs.com/cli/v8/using-npm/scripts#pre--post-scripts) for this kind of thing so they can automatically update the json when running on localhost or when building locally. [Middleware](https://vercel.com/docs/concepts/functions/edge-middleware) is another option for rewriting and redirecting to the correct API dynamically.
